docs/developer-feature-tutorial.md

```diff
@@ -99,74 +165,113 @@ Now, run zulip terminal and open web app from a different account. Start composi
 }
 ```
 
-You can view these events in the `type` file in your `zulip-terminal` home directory.
+You can view these events in the `type_test.json` file in your `zulip-terminal` home directory.
+
+Now to display if a user is typing in the View, we need to check few things:
 
-Now to display if user is typing in the view, we need to check few things:
 * The `op` is `start`.
 * User is narrowed into PM with a user.
-* The `user_id` of the person is present in the narrowed PM recipients.
+* The `user_id` of the sender is present in the PM narrow of the user.
 
-If all the above conditions are satisfied we can successfully update the footer to display `X is typing` until we receive
-a `stop` event for typing.
+If all the above conditions are satisfied, we can successfully update the footer to display `<sender> is typing...` until we receive a `stop` event for typing.
 
-To check for the above conditions, we create a function in `ui.py`:
-```python
+To check for the above conditions, we create a method in the Model class in `model.py`:
 
-    def handle_typing_event(self, event: Dict['str', Any]) -> None:
+```python
+# zulipterminal/model.py
+# In Model class
+def _handle_typing_event(self, event: Event) -> None:
+    """
+    Handle typing notifications (in private messages)
+    """
+    assert event['type'] == "typing"
+    if hasattr(self.controller, 'view'):
         # If the user is in pm narrow with the person typing
-        if len(self.model.narrow) == 1 and\
-                self.model.narrow[0][0] == 'pm_with' and\
-                event['sender']['email'] in self.model.narrow[0][1].split(','):
+        narrow = self.narrow
+        if (len(narrow) == 1 and narrow[0][0] == 'pm_with' and event['sender']['email'] in narrow[0][1].split(',')):
```

Comment:
This part is different already, and #990 will update it further (though it can be updated with/after that PR later)